### PR TITLE
Analyzer - Generate baseline given results from multiple nodes. 

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Generate baseline given results from multiple nodes. 

**Major Revision**
- Add sub command `sb result generate-baseline`
- Add UT and docs

### CODE REVIEW

**Review by cp5555 — APPROVED** [review]

File: docs/cli.md [hunk 1]
```diff
@@ -275,6 +275,53 @@ Run result summary and output the results in html format:
 sb result summary --data-file outputs/results-summary.jsonl --rule-file rule.yaml --output-file-format html
 ```
 
+### `sb result generate-baseline`
+
+Generate the baseline file automatically from multiple machines results according to rules defined in rule file.
+
+```bash title="SB CLI"
+sb result summary --data-file
```
Comment (yukirora) [inline docs/cli.md:283]: sb result generate-baseline?

File: docs/user-tutorial/baseline-generation.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+id: baseline-generation
```
Comment (yukirora) [inline docs/user-tutorial/baseline-generation.md:2]: please add this page into website/sidebars.js

File: superbench/analyzer/baseline_generation.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                    elif isinstance(self._raw_data_df[metric].iloc[0], int):
+                        baseline[metric] = int(val)
+                    else:
+                        try:
+                            baseline[metric] = float(val)
+                        except Exception as e:
+                            logger.error('Analyzer: {} baseline is not numeric, msg: {}'.format(metric, str(e)))
+            baseline = json.dumps(baseline, indent=2, sort_keys=True)
+            baseline = re.sub(r': \"(\d+.?\d*)\"', r': \1', baseline)
+            with open(output_dir + '/baseline.json', mode='w') as f:
```
Comment (yukirora) [inline superbench/analyzer/baseline_generation.py:189]: please help revise this to Path(output_dir) / 'baseline.json'

File: docs/cli.md (hunk 1 above)
Comment (guoshzhao) [reply docs/cli.md:283]: Good catch. Fixed.

File: docs/user-tutorial/baseline-generation.md (hunk 2 above)
Comment (guoshzhao) [reply docs/user-tutorial/baseline-generation.md:2]: Done.

File: superbench/analyzer/baseline_generation.py (hunk 3 above)
Comment (guoshzhao) [reply superbench/analyzer/baseline_generation.py:189]: Done.

**Comment by yukirora** [comment]
@guoshzhao , could u please solve the UT failure `ERROR: Could not find a version that satisfies the requirement numpy>=1.20.3 (from superbench[cpuworker,test]) (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0, 1.13.1, 1.13.3, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5)
ERROR: No matching distribution found for numpy>=1.20.3
WARNING: You are using pip version 21.2.4; however, version 21.3.1 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.`

**Review by yukirora — APPROVED** [review]

**Pull request merged**